File: Olympiad/COCI/official/2007/contest3_solutions/trojke.cpp

/*
  Croatian Open Competition in Informatics 2006/2007
  Contest 3 - Task TROJKE
  Programming language C++
*/

#include <cstdio>

char a[100][100+1];

int n, m;
int x[26], y[26];

int main( void ) {
   scanf( "%d", &n );

   for( int i = 0; i < n; ++i ) {
      scanf( "%s", a[i] );
      for( int j = 0; j < n; ++j ) {
         if( a[i][j] == '.' ) continue;
         x[m] = i;
         y[m] = j;
         ++m;
      }
   }

   int rjesenje = 0;

   for( int i = 0; i < m; ++i )
      for( int j = i+1; j < m; ++j )
         for( int k = j+1; k < m; ++k )
            if( x[i]*(y[j]-y[k]) + x[j]*(y[k]-y[i]) + x[k]*(y[i]-y[j]) == 0 )
               ++rjesenje;

   printf( "%d\n", rjesenje );

   return 0;
}

File: Olympiad/COCI/official/2007/contest3_solutions/trojke.pas

/*
  Croatian Open Competition in Informatics 2006/2007
  Contest 3 - Task TROJKE
  Programming language Pascal
*/

program trojke;

var
   a : array [1..100,1..101] of char;
   n, m : integer;
   i, j, k : integer;
   x, y : array [1..26] of integer;
   rjesenje : integer;

begin
   readln( n );

   m := 0;
   for i := 1 to n do begin
      readln( a[i] );
      for j := 1 to n do begin
         if a[i,j] = '.' then continue;
         m := m + 1;
         x[m] := i;
         y[m] := j;
      end;
   end;

   rjesenje := 0;
   for i := 1 to m do
      for j := i+1 to m do
         for k := j+1 to m do
            if x[i]*(y[j]-y[k]) + x[j]*(y[k]-y[i]) + x[k]*(y[i]-y[j]) = 0 then
               rjesenje := rjesenje + 1;
   writeln( rjesenje );
end.
